hera_cal/nucal.py

```diff
@@ -0,0 +1,473 @@
+            if len(pols) > 1:
+                raise ValueError(
+                    f"Multiple polarizations are in your radially redundant group: {pols}"
+                )
+            else:
+                self.pol = pols[0]
+        else:
+            self.pol = pol
+
+        if blvec is None:
+            ant1, ant2, pol = baselines[0]
+            self.blvec = (antpos[ant2] - antpos[ant1]) / np.linalg.norm(
+                antpos[ant2] - antpos[ant1]
+            )
+        else:
+            self.blvec = blvec
+
+        # Store baseline lengths
+        baseline_lengths = []
+        for baseline in baselines:
+            ant1, ant2, pol = baseline
+            baseline_lengths.append(np.linalg.norm(antpos[ant2] - antpos[ant1]))
+
+        # Sort baselines list by baseline length
+        self._baselines = [_baselines[idx] for idx in np.argsort(baseline_lengths)]
+        self.baseline_lengths = [
+            baseline_lengths[idx] for idx in np.argsort(baseline_lengths)
+        ]
+
+    def get_u_bounds(self, freqs):
+        """
+        Calculates the magnitude of the minimum and maximum u-modes values of the radial redundant group
+        given an array of frequency values
+
+        Parameters:
+        ----------
+        freqs: np.ndarray
+            Array of frequencies found in the data in units of Hz
+
+        Returns:
+        -------
+        ubounds: tuple
+            Tuple of the magnitude minimum and maximum u-modes sampled by this baseline group
+        """
+        umin = freqs.min() / SPEED_OF_LIGHT * np.min(self.baseline_lengths)
+        umax = freqs.max() / SPEED_OF_LIGHT * np.max(self.baseline_lengths)
+        return (umin, umax)
+
+    def filter_group(
+        self,
+        bls=None,
+        ex_bls=None,
+        ants=None,
+        ex_ants=None,
+        pols=None,
+        ex_pols=None,
+        min_bl_cut=None,
+        max_bl_cut=None,
+    ):
+        """
+        Filter radially redundant group to include/exclude the specified bls, antennas. and polarizations.
+        Arguments are evaluated, in order of increasing precedence: (pols, ex_pols, bls, ex_bls, ants, ex_ants,
+        min_bl_cut, max_bl_cut).
+
+        Parameters:
+        ----------
+        bls : list of tuples, default=None
+            baselines to include. Baselines of the form (i,j,pol) include that specific
+            visibility.  Baselines of the form (i,j) are broadcast across all polarizations present in reds.
+        ex_bls : list of tuples, default=None
+            same as bls, but excludes baselines.
+        ants : list of tuples, default=None
+            antennas to include. Only baselines where both antenna indices are in ants
+            are included.  Antennas of the form (i,pol) include that antenna/pol. Antennas of the form i are
+            broadcast across all polarizations present in reds.
+        ex_ants : list of tuples, default=None
+            same as ants, but excludes antennas.
```

Comment:
I think it's confusing to offer this option, because usually these objects will be constructed with unique baselines, in which case filtering by antennas is potentially misleading. If you really want this to be possible, I'd suggest adding a `**filter_kwargs` which passes into redcal.filter_reds and lets you do other options you don't explicitly call out here.

Reply:
I just decided to simplify this and only include filtering by baseline length and number of unique baselines per radially redundant group